### ISSUE TITLE
URL parsing failures with long URLs

### ISSUE THREAD

**Issue by nemobis** [connectbot/connectbot#343]
Rarely, parsing of URLs failed for me. I did not manage to make a self-explanatory screenshot, but one issue is: when the URL ends at the very last character of the line, the first word of the next line is appended to the URL.

It also seems that everything before a colon is considered a possible protocol, but that may be intended to avoid having an hardcoded list of protocols.
https://up1.ca/#lM_Nj8kGpTY4DjrxDuGlSQ

**Comment by leonerd** [comment]
Yeah; that's a really hard problem to solve. If the URL "ends" on the last cell of a line, it's impossible to tell if it was supposed to continue wrapped into the next line or not.

**Comment by belg4mit** [comment]
I find this an annoying misfeature that marrs an otherwise wonderful tool. What about lines that are soft-wrapped, it seems like those should be trivial to detect, yet it does not.

What if one pre-selected a region, and URL-scan could then know to find the longestreasonable URL in the selection?
